django/db/models/expressions.py

```diff
@@ -843,13 +843,19 @@ def as_sql(self, compiler, connection, template=None, case_joiner=None, **extra_
         template_params.update(extra_context)
         case_parts = []
         sql_params = []
+        from django.db.models.sql.datastructures import EmptyResultSet
```

Comment:
Good point, I didn't think of that module as being user facing. Perhaps django.db.utils would be a better place? It already contains various DatabaseError exceptions, so it wouldn't be totally out of place.